Steven/refactor

jkteng opened the pull request.

Commit:
log-in modal and username local storage added

jkteng committed:
Sign-in modal, username local storage, and logout button added

jkteng committed:
Merge branch 'refactor' of github.com:innovap3/js-exam into steven/refactor

jkteng committed:
[https://trello.com/c/ARyGE5xH] feat: Sets Interviewer name when user first load page

- remove aws authentication forced login

jkteng committed:
delete .editorconfig

jkteng committed:
[https://trello.com/c/ARyGE5xH] feat: Sets Interviewer name when user first load page

- redirect to /admin whenever logged out

pohsiangchen commented on src/containers/MainPage/SignOnModal/index.js:
```diff
@@ -0,0 +1,92 @@
+    const username = this.state.username;
+    try {
+      localStorage.setItem('username', username);
+      console.log('User: ', username);
```
remove this line

pohsiangchen commented on src/redux/login/reducer.js:
```diff
@@ -1,10 +1,19 @@
+const login = (state = initialState, action) => {
   switch (action.type) {
     case 'LOGIN/LOGIN': {
       return { isLogin: authLogin(action.password) };
```
Should change to `{ ...state, isLogin: authLogin(action.password) }`

pohsiangchen commented on src/redux/login/reducer.js:
```diff
@@ -1,10 +1,19 @@
       return { isLogin: authLogin(action.password) };
     }
+    case SET_USERNAME: {
+      return { username: action.data };
```
Should change to `{ ...state, username: action.data }`

pohsiangchen commented on src/components/Widgets/TabWidget/index.js:
```diff
@@ -57,8 +68,46 @@ const TabWidget = ({
+
+const withConnect = connect(
+  mapStateToProps,
+  null,
```
Should delete

pohsiangchen commented on src/containers/MainPage/SignOnModal/index.js:
```diff
@@ -0,0 +1,92 @@
+  }
+
+  signedOn = () => {
+    if (localStorage.getItem('username')) return true;
```
Here we can use `username` from reducer `login -> username` to check if we already signed on.

jkteng committed:
[https://trello.com/c/ARyGE5xH] feat: Sets Interviewer name when user first load page

- move SignOnModal from MainPage to Admin

jkteng committed:
minor fix in route.js

jkteng committed:
[https://trello.com/c/ARyGE5xH] feat: Sets Interviewer name when user first load page

- username state mapped to props in Tabwidget
- Tabwidget switched to class component
